### PR TITLE
fix: propagate edge zones to private endpoints

### LINKED ISSUE #365: [AVM Module Issue]: Edge Zones Unsupported for Private Endpoints
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### (Optional) Module Version

Appears to affect all modules including latest 0.7.3

### (Optional) Correlation Id

_No response_

### Description

When deploying the module with the parameter:

edge_zone = "perth"

and the private_endpoint parameter set:

  private_endpoints = {
    "pe" = {
      name                   = "${var.storage_account_name}-pe"
      subnet_resource_id     = data.azurerm_subnet.existing.id
      subresource_name       = "blob"
      network_interface_name = "${var.storage_account_name}-pe-nic"
    }
  }

An apply of the module returns: 

│ RESPONSE 400: 400 Bad Request
│ ERROR CODE: VnetSpanningForEdgeZonesNotEnabled

It appears that the module attempts to deploy the NIC in the parent location, rather than the edge zone location where the storage account has been deployed. I was able to temporarily workaround this problem by adding an extendedLocation block to the body of avm-res-storage-storageaccount/modules/private_endpoint/main.tf, but it seems that the latest version of this module lacks support for edge zones only in private endpoints:

extendedLocation = {
      name = "perth"
      type = "EdgeZone"
    }

### PR DESCRIPTION
**Pull request by myaschmitz**
## Problem

When `edge_zone = "perth"` is set, the storage account request includes `body.extendedLocation`, but the private endpoint child module previously received no Edge Zone value. Its `Microsoft.Network/privateEndpoints` request therefore omitted `extendedLocation`, and Azure rejected the cross-placement network operation with `400 VnetSpanningForEdgeZonesNotEnabled`.

## Change

- Pass the root storage account `edge_zone` into every managed private endpoint automatically.
- Emit top-level `body.extendedLocation = { name = <zone>, type = "EdgeZone" }` in the private endpoint AzAPI request when configured.
- Merge the optional field into the request so ordinary regional private endpoints omit `extendedLocation` entirely rather than sending `null`.
- Add mock tests for root Edge Zone propagation and regional omission.
- Regenerate root and private endpoint submodule documentation through AVM tooling.

A per-endpoint override was intentionally not added because the current AVM standard `private_endpoints` interface requires an exact object schema and rejects additional fields. Automatic inheritance fixes the reported configuration without redundant caller input and preserves interface compliance.

## Validation

- `./avm tf-test-unit` — passed
- `./avm pre-commit` — passed
- `./avm pr-check` — formatting, validation, docs, unit tests, and TFLint passed; local Well-Architected example plans could not complete because the Azure CLI token is absent on this machine and will run with credentials in CI

## Testing scope

Coverage for the Edge Zone path is by mock unit test rather than a deployed example, and this is deliberate.

The tests assert the exact request shape (`{ name = "perth", type = "EdgeZone" }`), assert regional endpoints omit `extendedLocation` entirely, and assert the storage account and its private endpoint carry the *same* placement - the divergence that produces `VnetSpanningForEdgeZonesNotEnabled`. Reverting the one-line propagation in `main.privateendpoint.tf` was verified to fail these tests, so they guard against regression rather than restating the implementation.

An `examples/` deployment was considered and rejected: Azure Extended Zones require `Microsoft.EdgeZones` provider registration plus per-subscription access approval and per-zone registration. Since `test-examples` applies every directory under `examples/` with no opt-out, an Edge Zone example would hard-fail CI on any subscription lacking that grant.

Closes #365

### CODE REVIEW

File: modules/private_endpoint/outputs.tf [hunk 1]
```diff
@@ -3,6 +3,11 @@ output "name" {
   value       = azapi_resource.this.name
 }
 
+output "extended_location" {
+  description = "The extended location sent in the private endpoint request, or `null` when the private endpoint is regional."
+  value       = try(azapi_resource.this.body.extendedLocation, null)
+}
```
Comment (jtracey93) [inline modules/private_endpoint/outputs.tf:9]: Is this being used by the root module? If not, probably not worth exposing

**Review by jtracey93 — DISMISSED** [review]

File: modules/private_endpoint/outputs.tf (hunk 1 above)
Comment (myaschmitz) [reply modules/private_endpoint/outputs.tf:9]: Good catch - you were right, nothing consumed it. It existed because `terraform test` can't reach into child-module internals, so the root-level assertion that the edge zone actually reaches the PE request needed a window into it. It's now surfaced on the root `private_endpoints` output as an `edge_zone` string, so it earns its place.

**Review by jtracey93 — APPROVED** [review]
